src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -438,6 +438,57 @@ static Value builtinRegExpReplace(ExecutionState& state, Value thisValue, size_t
     return Value(builder.finalize(&state));
 }
 
+static Value builtinRegExpMatch(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    Value rx = thisValue;
+
+    if (!rx.isObject()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().RegExp.string(), true,
+                                       state.context()->staticStrings().toPrimitive.string(), errorMessage_GlobalObject_ThisNotObject);
+    }
+
+    String* str = argv[0].toString(state);
+    Value arg[1] = { str };
+    ASSERT(str != nullptr);
+
+    //21.2.5.6.8
+    bool global = rx.asObject()->get(state, ObjectPropertyName(state, state.context()->staticStrings().global)).value(state, rx).toBoolean(state);
+
+    if (!global) {
+        return builtinRegExpExec(state, rx, 1, arg, false);
+    }
+
+    bool fullUnicode = rx.asObject()->get(state, ObjectPropertyName(state, state.context()->staticStrings().unicode)).value(state, rx).toBoolean(state);
+    rx.asObject()->set(state, ObjectPropertyName(state, state.context()->staticStrings().lastIndex), Value(0), rx);
```

Comment:
IMO `setThrowsException` is invoked here instead of `set` according to the [standard](https://www.ecma-international.org/ecma-262/6.0/#sec-regexp.prototype-@@match) which says when `set` operation fails, then TypeError exception should be thrown.